src/history.rs

```
@@ -669,169 +633,48 @@ impl HistoryImpl {
 
     /// Saves history by rewriting the file.
     fn save_internal_via_rewrite(&mut self) -> std::io::Result<()> {
-        // We want to rewrite the file, while holding the lock for as briefly as possible
-        // To do this, we speculatively write a file, and then lock and see if our original file changed
-        // Repeat until we succeed or give up
-        let Some(possibly_indirect_target_name) = history_filename(&self.name, L!(""))? else {
+        let Some(history_path) = self.history_file_path()? else {
             return Ok(());
         };
-        let Some(tmp_name_template) = history_filename(&self.name, L!(".XXXXXX"))? else {
-            return Err(std::io::Error::new(
-                std::io::ErrorKind::Other,
-                "In this context the `None` variant should never occur.",
-            ));
-        };
```

range-diff for `988648e82597d64fe9f52d78a40728ef2d12a778...09b8a1a447d8d877aa544069124d715eaf0e6b9f` looks good so far.

Two comments here:
1. `In this context the `None` variant should never occur` this means that instead of returning an error, we should crash (i.e. use `unwrap()`). This assertion helps the reader, and we can afford checking all callers. I'd probably also remove the error message, since it's fairly easy to check that the assertion holds
2. This commit actually changes the `Error(...)` to `Ok(())`, which is surprising. It doesn't matter since it never happens but better use `history_filename(...)?.unwrap()` throughout

---

I fully agree with your first point, that can just be an unwrap.

I don't know what you are referring to in your second point. Which commit and exact location do you mean?

---

That's a good change. Added.